modules/core/04-channel/keeper/upgrade.go

```diff
@@ -109,6 +110,110 @@ func (k Keeper) WriteUpgradeTryChannel(
 	emitChannelUpgradeTryEvent(ctx, portID, channelID, channel, proposedUpgrade)
 }
 
+// startFlushUpgradeHandshake will verify the counterparty proposed upgrade and the current channel state.
+// Once the counterparty information has been verified, it will be validated against the self proposed upgrade.
+// If any of the proposed upgrade fields are incompatible, an upgrade error will be returned resulting in an
+// aborted upgrade.
+//
+//lint:ignore U1000 Ignore unused function temporarily for debugging
+func (k Keeper) startFlushUpgradeHandshake(
```

Comment:
My understanding is that the writes that happen in the spec will happen in a separate function in the implementation?

Reply:
they will happen in the `WriteUpgradeInit/Try/Ack` etc functions